src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -0,0 +1,557 @@
+            {
+                // The new name differs from the old name,
+                // update our name.
+                auto oldCmdImpl{ get_self<Command>(oldCmd) };
+                oldCmdImpl->Name(newName);
+            }
+        }
+
+        // The same goes for consolidated actions.
+        if (consolidatedCmd)
+        {
+            // This command has a name, check if it's new.
+            if (newName != consolidatedCmd.Name())
+            {
+                // The new name differs from the old name,
+                // update our name.
+                auto consolidatedCmdImpl{ get_self<Command>(consolidatedCmd) };
+                consolidatedCmdImpl->Name(newName);
+            }
+        }
+
+        // Handle a collision with NestedCommands
+        _NestedCommands.TryRemove(newName);
+    }
+
+    // Method Description:
+    // - Update our internal state with the key chord of the newly registered action
+    // Arguments:
+    // - cmd: the action we're trying to register
+    // - oldCmd: the action that already exists in our internal state. May be null.
+    // - consolidatedCmd: the consolidated action that already exists in our internal state. May be null.
+    void ActionMap::_TryUpdateKeyChord(const Model::Command& cmd, const Model::Command& oldCmd, const Model::Command& consolidatedCmd)
+    {
+        if (const auto keys{ cmd.Keys() })
+        {
+            const auto oldKeyPair{ _KeyMap.find(keys) };
+            if (oldKeyPair != _KeyMap.end())
+            {
+                // Collision: The key chord was already in use.
+                // Remove the old one.
+                const auto& actionPair{ _ActionMap.find(oldKeyPair->second) };
+                const auto& conflictingCmd{ actionPair->second };
+                const auto& conflictingCmdImpl{ get_self<implementation::Command>(conflictingCmd) };
+                conflictingCmdImpl->EraseKey(keys);
+            }
+            else if (const auto& conflictingCmd{ GetActionByKeyChord(keys) })
+            {
+                // Collision: The key chord was already in use, but by an action in another layer
+                const auto conflictingActionID{ ActionHash{}(conflictingCmd.ActionAndArgs()) };
+                const auto& consolidatedCmdPair{ _ConsolidatedActions.find(conflictingActionID) };
+                if (consolidatedCmdPair != _ConsolidatedActions.end())
+                {
+                    const auto& consolidatedCmdImpl{ get_self<implementation::Command>(consolidatedCmdPair->second) };
+                    consolidatedCmdImpl->EraseKey(keys);
+                }
+                else
+                {
+                    // Create a copy of the conflicting action,
+                    // and erase the conflicting key chord from the copy.
+                    const auto& conflictingCmdImpl{ get_self<implementation::Command>(conflictingCmd) };
+                    const auto& conflictingCmdCopy{ conflictingCmdImpl->Copy() };

```

Comment:
i think for almost all of these what you really want is `auto` (not `const auto&`)

Reply:
the compiler can figure it out because `const auto&` is special, but the reader needs some help :)